sdcm/cluster_aws.py

```diff
@@ -417,20 +419,52 @@ def node_config_setup(self, node, seed_address, endpoint_snitch):
                               client_encrypt=self._param_enabled('client_encrypt'),
                               append_scylla_args=self.params.get('append_scylla_args'))
         else:
-            node.config_setup(enable_exp=self._param_enabled('experimental'),
-                              endpoint_snitch=endpoint_snitch,
-                              authenticator=self.params.get('authenticator'),
-                              server_encrypt=self._param_enabled('server_encrypt'),
-                              client_encrypt=self._param_enabled('client_encrypt'),
-                              append_scylla_args=self.params.get('append_scylla_args'))
+            if self.params.get('stop_service', default='true') == 'true':
+                node.config_setup(seed_address=seed_address,
+                                  enable_exp=self._param_enabled('experimental'),
+                                  endpoint_snitch=endpoint_snitch,
+                                  authenticator=self.params.get('authenticator'),
+                                  server_encrypt=self._param_enabled('server_encrypt'),
+                                  client_encrypt=self._param_enabled('client_encrypt'),
+                                  append_scylla_args=self.params.get('append_scylla_args'))
+            else:
+                node.config_setup(enable_exp=self._param_enabled('experimental'),
+                                  endpoint_snitch=endpoint_snitch,
+                                  authenticator=self.params.get('authenticator'),
+                                  server_encrypt=self._param_enabled('server_encrypt'),
+                                  client_encrypt=self._param_enabled('client_encrypt'),
+                                  append_scylla_args=self.params.get('append_scylla_args'))
 
     def node_setup(self, node, verbose=False, timeout=3600):
         endpoint_snitch = self.params.get('endpoint_snitch')
-        seed_address = self.get_seed_nodes_by_flag(private_ip=False)
+        if len(self.datacenter) > 1:
+            seed_address = self.get_seed_nodes_by_flag(private_ip=False)
+        else:
+            seed_address = self.get_seed_nodes_by_flag(private_ip=True)
+
+        def scylla_ami_setup_done():
+            """
+            Scylla-ami-setup will update config files and operate the scylla-server service.
+            We use this function to make sure scylla-ami-setup finishes, and first start is
+            done (fail as expected, /etc/scylla/ami_disabled is deleted). Then it won't effect
+            reconfig in SCT.
+
+            # scylla-ami-setup finishes:
```

Comment:
why are these commented out?

Reply:
Sorry for the misunderstand. The two commented code, and the real function code are 3 different opportunity.
The two commented code is easy to understand, but the last is really working. 

--stop-services parameter in ec2 user-data, not really stop running scylla-server service. But deleting a flag file (/etc/scylla/ami_disabled) in first start of scylla-server (by scylla_prepare), and fail the first start.

So we need to wait the first start fail by scylla_ami_setup_done().

Reply:
Maybe we can use different name for this helper func?
such as, wait_first_start_fail() / wait_first_start_exit()